hydrator-transforms-lib/src/main/java/co/cask/hydrator/transforms/PythonEvaluator.java

```diff
@@ -0,0 +1,350 @@
+      "a dictionary that represents the transformed input. " +
+      "For example:\n" +
+      "'def transform(input, emitter, context):\n" +
+      "  input['count'] *= 1024\n" +
+      "  if(input['count'] < 0):\n" +
+      "    context.getMetrics().count(\"negative.count\", 1)\n" +
+      "    context.getLogger().debug(\"Received record with negative count\")\n" +
+      "  emitter.emit(input)'\n" +
+      "will scale the 'count' field by 1024.")
+    private final String script;
+
+    @Description("The schema of output objects. If no schema is given, it is assumed that the output schema is " +
+      "the same as the input schema.")
+    @Nullable
+    private final String schema;
+
+    public Config(String script, String schema) {
+      this.script = script;
+      this.schema = schema;
+    }
+  }
+
+  // for unit tests, otherwise config is injected by plugin framework.
+  public PythonEvaluator(Config config) {
+    this.config = config;
+  }
+
+  @Override
+  public void initialize(TransformContext context) {
+    metrics = context.getMetrics();
+    logger = LoggerFactory.getLogger(PythonEvaluator.class.getName() + " - Stage:" + context.getStageId());
+    init();
+  }
+
+  @Override
+  public void destroy() {
+    if (interpreter != null) {
+      interpreter.cleanup();
+    }
+  }
+
+  /**
+   * Emitter to be used from within Python code
+   */
+  public final class PythonEmitter implements Emitter<Map> {
+
+    private final Emitter<StructuredRecord> emitter;
+    private final Schema schema;
+
+    public PythonEmitter(Emitter<StructuredRecord> emitter, Schema schema) {
+      this.emitter = emitter;
+      this.schema = schema;
+    }
+
+    @Override
+    public void emit(Map value) {
+      emitter.emit(decode(value));
+    }
+
+    @Override
+    public void emitError(InvalidEntry<Map> invalidEntry) {
+      emitter.emitError(new InvalidEntry<>(invalidEntry.getErrorCode(), invalidEntry.getErrorMsg(),
+                                           decode(invalidEntry.getInvalidRecord())));
+    }
+
+    public void emitError(Map invalidEntry) {
+      emitter.emitError(new InvalidEntry<>((int) invalidEntry.get("errorCode"),
+                                           (String) invalidEntry.get("errorMsg"),
+                                           decode((Map) invalidEntry.get("invalidRecord"))));
+    }
+
+    private StructuredRecord decode(Map nativeObject) {
+      return decodeRecord(nativeObject, schema);
+    }
+  }
+
+  @Override
+  public void transform(StructuredRecord input, final Emitter<StructuredRecord> emitter) {
+    try {
+      Emitter<Map> pythonEmitter = new PythonEmitter(emitter, schema == null ? input.getSchema() : schema);
+      interpreter.set(INPUT_STRUCTURED_RECORD_VARIABLE_NAME, encode(input, input.getSchema()));
+      interpreter.set(EMITTER_VARIABLE_NAME, pythonEmitter);
+      Py.runCode(compiledScript, interpreter.getLocals(), interpreter.getLocals());
```

Comment:
The third parameter has to be global variables, right? If we don't have anything, can we make it empty?